wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/notification/PreProjectInitializedEvent.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.project.server.notification;
+
+import org.eclipse.che.api.core.model.workspace.config.ProjectConfig;
+import org.eclipse.che.api.core.notification.EventOrigin;
+
+/**
+ * Publish before project initialization.
+ *
+ * @author Valeriy Svydenko
+ */
+@EventOrigin("project")
+public class PreProjectInitializedEvent {
```

Comment:
Please consider `BeforeProjectInitializationEvent` or `ProjectInitializationEvent`.

Reply:
Why not `PreProjectInitializedEvent`?

Reply:
@mmorhun refers to existed events. All of them start with `before` prefix.